pkg/cloud/services/networking/network.go

```diff
@@ -148,19 +147,17 @@ func (s *Service) ReconcileNetwork(openStackCluster *infrav1.OpenStackCluster, c
 	return nil
 }
 
-func (s *Service) DeleteNetwork(openStackCluster *infrav1.OpenStackCluster, network *infrav1.Network) error {
-	if network == nil || network.ID == "" {
-		s.logger.V(4).Info("No need to delete network since no network exists.")
-		return nil
-	}
-	exists, err := s.existsNetwork(network.ID)
+func (s *Service) DeleteNetwork(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
+	networkName := getNetworkName(clusterName)
+	network, err := s.getNetworkByName(networkName)
 	if err != nil {
 		return err
 	}
-	if !exists {
-		s.logger.Info("Skipping network deletion because network doesn't exist", "network", network.ID)
+	if network.ID == "" {
```

Comment:
```suggestion
	if network == nil || network.ID == "" {
```

Reply:
Can we return a Network pointer in getNetworkByName? Would be more consistent with our other code

Reply:
Our code is not consistent as you expect. functions in networking package do not return pointer. functions in compute and loadbalancer do return pointer.  If we update getNetworkByName, it is inconsistent in networking package.

Reply:
Okay let's leave it like it is for now. Although I want to fix this sooner or later, but we don't have to do it here.